force_wfmanager/left_side_pane/tests/test_workflow_model_view.py

```diff
@@ -51,3 +63,18 @@ def test_remove_notification_listener(self):
         self.assertEqual(len(self.wf_mv.notification_listeners_mv), 1)
         self.wf_mv.remove_notification_listener(model)
         self.assertEqual(len(self.wf_mv.notification_listeners_mv), 0)
+
+    def test_remove_datasource(self):
+        self.wf_mv_name_registry.add_execution_layer(ExecutionLayer())
+        self.assertEqual(len(self.wf_mv_name_registry.
+                             execution_layers_mv[0].model.data_sources), 0)
+        self.wf_mv_name_registry.execution_layers_mv[0].\
```

Comment:
Use a local variable here - if you are splitting a line on a `.` that's an indicator that you are chaining too deep.

If you do something like
```
execution_layer = self.wf_mv_name_registry.execution_layers_mv[0]
```
then I think that not only will this line be a lot cleaner, but so will subsequent lines.